srml/staking/src/lib.rs

```diff
@@ -552,54 +552,66 @@ decl_module! {
 
 			let mut ledger = Self::ledger(&controller).ok_or("not a controller")?;
 			let StakingLedger {
+				stash,
+				total_ring,
 				active_ring,
 				active_deposit_ring,
+				total_kton,
 				active_kton,
-				unlocking,
 				..
 			} = &mut ledger;
 
 			ensure!(
-				unlocking.len() < MAX_UNLOCKING_CHUNKS,
+				T::Ring::locks_count(stash) + T::Kton::locks_count(stash) < MAX_UNLOCKING_CHUNKS,
 				"can not schedule more unlock chunks"
 			);
 
-			let era = Self::current_era() + T::BondingDuration::get();
-
+			let at = <timestamp::Module<T>>::now().saturated_into::<TimeStamp>() + T::BondingDuration::get();
```

Comment:
Check: why change from ```Self::current_era() ``` to ```<timestamp::Module<T>>::now()```

Reply:
The lock require a `TimeStamp` to chcek if expired.